libr/core/cmd_open.c

```diff
@@ -1422,6 +1423,15 @@ static int cmd_open(void *data, const char *input) {
 					addr = UT64_MAX;
 				}
 				r_core_bin_load (core, argv0, addr);
+				if (*input == '+') { // "o+"
+					SdbListIter *iter;
+					RIOMap *map;
+					ls_foreach_prev (core->io->maps, iter, map) {
+						if (map->fd == fd) {
+							map->perm |= R_PERM_WX;
```

Comment:
if the underlying fd doesnt permit W this will not be correct

Reply:
ok bcbec03e5cb25a72264e8ac7c702d9d38ffe185c ... btw, tried this with a `chmod -w` bin and the error actually happened in `r_core_file_open()`, but the extra check appears harmless